src/Halogen/DatePicker/Component/Date.purs

```diff
@@ -0,0 +1,182 @@
+module Halogen.Datapicker.Component.Date where
+
+import Prelude
+
+import Data.Bifunctor (bimap)
+import Data.Date (Date, Day, Month, Year)
+import Data.Either (Either(..))
+import Data.Either.Nested (Either2)
+import Data.Enum (class BoundedEnum, fromEnum, toEnum, upFromIncluding)
+import Data.Functor.Coproduct (Coproduct, coproduct, right, left)
+import Data.Functor.Coproduct.Nested (Coproduct2)
+import Data.Generic.Rep (class Generic)
+import Data.Generic.Rep.Show (genericShow)
+import Data.Array (sort)
+import Data.Maybe (Maybe(..), maybe)
+import Data.Monoid (mempty)
+import Data.Newtype (unwrap)
+import Data.NonEmpty (NonEmpty)
+import Data.Profunctor.Join (Join(..))
+import Data.Profunctor.Star (Star(..))
+import Data.Traversable (fold, for, sequence)
+import Halogen as H
+import Halogen.Component.ChildPath as CP
+import Halogen.Datapicker.Format.Date as F
+import Halogen.Datapicker.Internal.Choice as Choice
+import Halogen.Datapicker.Internal.Elements (textElement)
+import Halogen.Datapicker.Internal.Enums (MonthShort, Year2, Year4, setYear)
+import Halogen.Datapicker.Internal.Num as Num
+import Halogen.Datapicker.Internal.Range (Range, bottomTop)
+import Halogen.Datapicker.Internal.Utils (steper', pickerClasses, mustBeMounted)
+import Halogen.Datapicker.Component.Types (BasePickerQuery(..), PickerMessage(..), PickerQuery(..), PickerValue, value)
+import Halogen.HTML as HH
+import Halogen.HTML.Events as HE
+import Halogen.HTML.Properties as HP
+
+data DateQuery a = Update (Date -> Maybe Date) a
+type Input = PickerValue DateError Date
+type QueryIn = PickerQuery Unit Input
+type Query = Coproduct QueryIn DateQuery
+type Message = PickerMessage Input
+
+data DateError = InvalidDate
+derive instance dateErrorEq :: Eq DateError
+derive instance dateErrorOrd :: Ord DateError
+derive instance dateErrorGeneric :: Generic DateError _
+instance dateErrorShow :: Show DateError where
```

Comment:
Generic instance is just for Show. Show is used in Example and might be useful for debugging.